The function is defined in the source, so don't inline it in the header.

--- arch/arm/include/asm/ftrace.h	2014-10-22 23:53:32.000000000 +0200
+++ arch/arm/include/asm/ftrace.h	2016-12-11 22:57:55.065090536 +0100
@@ -45,10 +45,7 @@
 
 #else
 
-extern inline void *return_address(unsigned int level)
-{
-	return NULL;
-}
+extern void *return_address(unsigned int level);
 
 #endif
 
